polaris-core/src/main/java/org/apache/polaris/core/policy/validator/PolicyValidator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.policy.validator;
+
+import org.apache.polaris.core.entity.PolarisEntitySubType;
+import org.apache.polaris.core.entity.PolarisEntityType;
+
+/**
+ * Validates and parses a given policy content string against its defined schema.
+ *
+ * @param <T> the type of policy object to be returned after validation
+ */
+public interface PolicyValidator<T> {
+
+  /**
+   * Parses and validates the provided policy content.
+   *
+   * @param content the policy content to parse and validate
+   * @return a policy object of type T if the content is valid
+   * @throws InvalidPolicyException if the content does not meet the required policy rules
+   */
+  T parse(String content) throws InvalidPolicyException;
```

Comment:
It looks like this is primarily a parser, not a validator if it's turning a string into a policy type.

Also, is there not a shared supertype for policies? Even a marker interface might be a good idea.

Reply:
> Also, is there not a shared supertype for policies? Even a marker interface might be a good idea.

Yeah, that's another option I thought of. I'm fine with either way. Are you suggesting to use a new policy interface to avoid type parameters(<T>)?

Reply:
I was imagining something like:

```
public interface Policy<T extends Policy<T>> {
    T fromString(String content);
}
```

This way I can just write my implementations like this:

```
public class CompactionPolicy implements Policy {
  public CompactionPolicy fromString(...)
}

public class RbacPolicy implements Policy {
  public RbacPolicy fromString(...)
}
```

Reply:
Introduce a marker interface. I didn't use this pattern  https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern, we may introduce it once we want to introduce the method `fromString()`. WDYT?

Reply:
> It looks like this is primarily a parser, not a validator if it's turning a string into a policy type.

Separated the parsing and validating in the new commit. The parsing is with the policy content class now.

Added a marker interface `PolicyContent`. `fromString()` is a static method, so I didn't use this pattern, https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern. We could have a pattern like `Policy<T extends Policy<T>>` once we need a inheritable method that could return a subtype.